addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/JsoScanner.java

```diff
@@ -0,0 +1,137 @@
+        checkJsoInHeaders(msg);
+
+        checkJsoInCookies(msg);
+
+        checkJsoInBody(msg);
+    }
+
+    private void checkJsoInBody(HttpMessage msg) {
+        byte[] startOfBody =
+                Arrays.copyOfRange(
+                        msg.getResponseBody().getBytes(), 0, JSO_BYTE_MAGIC_SEQUENCE.length);
+        String startOfBodyAsString =
+                new String(
+                        Arrays.copyOfRange(
+                                msg.getResponseBody().getBytes(),
+                                0,
+                                JSO_BASE_64_MAGIC_SEQUENCE.length()),
+                        StandardCharsets.UTF_8);
+        if (Arrays.equals(JSO_BYTE_MAGIC_SEQUENCE, startOfBody)
+                || hasJsoBase64MagicSequence(startOfBodyAsString)) {
+            Alert alert = buidAlert(msg, "");
+            parent.raiseAlert(getPluginId(), alert);
+        }
+    }
+
+    private void checkJsoInCookies(HttpMessage msg) {
+        for (HttpCookie cookie : msg.getResponseHeader().getHttpCookies(null)) {
+            String value = cookie.getValue();
+            if (!hasJsoBase64MagicSequence(value)) {
+                continue;
+            }
+
+            Alert alert = buidAlert(msg, cookie.toString());
+            parent.raiseAlert(getPluginId(), alert);
+        }
+    }
+
+    private boolean hasJsoBase64MagicSequence(String value) {
+        return value.startsWith(JSO_BASE_64_MAGIC_SEQUENCE);
+    }
+
+    private void checkJsoInHeaders(HttpMessage msg) {
+        for (HttpHeaderField header : msg.getResponseHeader().getHeaders()) {
+            String value = header.getValue();
+            if (!hasJsoBase64MagicSequence(value)) {
+                continue;
+            }
+
+            Alert alert = buidAlert(msg, header.toString());
+            parent.raiseAlert(getPluginId(), alert);
+        }
+    }
+
+    private Alert buidAlert(HttpMessage msg, String evidence) {
+        Alert alert =
+                new Alert(getPluginId(), Alert.RISK_MEDIUM, Alert.CONFIDENCE_MEDIUM, getName());
+        alert.setDetail(
+                Constant.messages.getString(MESSAGE_PREFIX + "desc"),
+                msg.getRequestHeader().getURI() + "",
+                null,
+                "",
+                "",
+                getString("soln"),
+                getString("refs"),
+                evidence,
+                502, // CWE-502: Deserialization of Untrusted Data
+                -1, // No WASC-ID
+                msg);
+        return alert;
+    }
+
+    private String getString(String param) {
+        return Constant.messages.getString(MESSAGE_PREFIX + param, "");
+    }
+
+    @Override
+    public void setParent(PassiveScanThread parent) {
+        this.parent = parent;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public int getPluginId() {
+        return 90303;
```

Comment:
Please submit a separate PR to reserve a plugin ID:
https://github.com/zaproxy/zaproxy/blob/develop/docs/scanners.md

`90002` seems like a good option if you're set on a 90k number for some reason. Otherwise perhaps `10110`.

Reply:
Done in: https://github.com/zaproxy/zaproxy/pull/5489